presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -292,14 +295,13 @@ public void evaluateFinal(int groupId, BlockBuilder output)
         }
 
         @Override
-        public void processOrderBy(GroupByHash groupByHash)
+        public void prepareFinal()
         {
             pagesIndex.sort(orderByChannels, orderings);
-            Iterator<Page> pagesIterator = pagesIndex.getPages(orderByOutputChannels);
+            Iterator<Page> pagesIterator = pagesIndex.getSortedPages();
             pagesIterator.forEachRemaining(page -> {
-                Work<GroupByIdBlock> groupIds = groupByHash.getGroupIds(page);
-                verify(groupIds.process());
-                accumulator.addInput(groupIds.getResult(), page);
+                GroupByIdBlock groupIds = new GroupByIdBlock(groupCount, page.getBlock(page.getChannelCount() - 1));
+                accumulator.addInput(groupIds, page);
```

Comment:
Not sure it it's an issue in practice, but we should probably pass page without the last column (i.e., the groupIds block) to the accumulator.